src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -188,9 +200,8 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
                     continue;
                 }
 
-                // if the address changed, remember the contact we want to add a new address for    
-                Contact conOld = (listOld != null ? (Contact)listOld[i] : null);
-                if (isContactAddressChanged(con, conOld)) 
+                // if the contacts address changed or if the contact with an address moves to a new household remember the contact
+                if (isContactAddressChanged(con, conOld) || (acctChanged)) 
```

Comment:
ok, I see your adding the || acctChange will get the address created (listConCreateAddr), but wouldn't that be incorrect if the contact has no address?